developer-docs-site/docs/move/book/functions.md

```diff
@@ -521,44 +521,58 @@ fun foo() { return () }
 
 ## Inline Functions
 
-Inline functions are functions which are expanded at caller side instead
-of compiled into Move bytecode. This allows to save gas and may lead
-to faster execution. For example, one can define an inline function
+Inline functions are functions whose bodies are expanded in-place at the caller location during compile time.
```

Comment:
"in place" not "in-place"

Reply:
Yes.  In "in-place algorithm" the phrase "in-place" is functioning as an adjective (modifying algorithm), while "expanded in place" the phrase "in place" is a prepositional phrase modifying the verb "expanded".  Or maybe it's that "in-place" is hyphenated because it appears *before* the modified noun.

[Here's a possible explanation ](https://writing-rag.com/153/hyphen-or-not/)

The MLA Handbook seems to have a thorough description, but it's paywalled:

A compound adjective formed from a prepositional phrase is hyphenated. The same type of compound is unhyphenated when it appears after the noun it modifies.

[2.33–2.48: Hyphens - MLA Handbook Plus](https://mlahandbookplus.org/books/book/5/chapter/56460/Hyphens#:~:text=A%20compound%20adjective%20formed%20from%20a%20prepositional%20phrase%20is%20hyphenated.&text=on%2Dcampus%20activities-,The%20same%20type%20of%20compound%20is%20unhyphenated%20when,after%20the%20noun%20it%20modifies.)

Reply:
I meant the various uses of "in-place" in that web page, not just the title.

For example, this is taken verbatim from that webpage: "Quicksort operates in-place on the data to be sorted." This usage of "in-place" is no different from mine. I have seen such usage of "in-place" quite commonly in CS literature, which is likely where I picked it up.